src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -507,42 +590,60 @@ public Task RemoveLoginAsync(IUser user, string loginProvider, string providerKe
                 throw new ArgumentNullException(nameof(user));
             }
 
-            var externalLogins = ((User)user).LoginInfos;
-            if (externalLogins != null)
+            if (user is User su && su.LoginInfos != null)
             {
-                var item = externalLogins.FirstOrDefault(c => c.LoginProvider == loginProvider && c.ProviderKey == providerKey);
+                var item = su.LoginInfos.FirstOrDefault(c => c.LoginProvider == loginProvider && c.ProviderKey == providerKey);
                 if (item != null)
                 {
-                    externalLogins.Remove(item);
+                    su.LoginInfos.Remove(item);
                 }
             }
+
             return Task.CompletedTask;
         }
 
         #endregion IUserLoginStore<IUser>
 
         #region IUserClaimStore<IUser>
 
-        public Task<IList<Claim>> GetClaimsAsync(IUser user, CancellationToken cancellationToken)
+        public async Task<IList<Claim>> GetClaimsAsync(IUser user, CancellationToken cancellationToken)
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
-            return Task.FromResult<IList<Claim>>(((User)user).UserClaims.Select(x => x.ToClaim()).ToList());
```

Comment:
Whatever was populating the claims in this collection should have a default implementation in the users module for the administrator role. It could also just assign ALL the permissions (not checking for Administrator).

Another is to have a custom handler (IAuthorizationHandler) that always allow any permission (only for the SuperUser), even if the Claims don't have permissions (empty list). If the implementation of RoleStore is the one from this module (NullRoleStore).

In case the Roles modules is inadvertently disabled, we need to only grant a permission to the SuperUser.